app/controllers/posts_controller.rb

```diff
@@ -8,4 +8,28 @@ def show
     @user = User.find(params[:user_id])
     @post = @user.posts.find(params[:id])
   end
+
+  def new
+    @user = User.find(params[:user_id])
+    @post = Post.new
+    respond_to do |format|
+      format.html { render :new, locals: { post: @post } }
+    end
+  end
+
+  def create
+    post_params = params.require(:post).permit(:title, :text)
```

Comment:
- [OPTIONAL] Just as I suggested above, you can consider extracting the `post_params` into its separate private function. It's going to make your controller clean and easy to understand. :+1: Here's a snippet

```ruby
    private 
    def post_params
      params.require(:post).permit(:title, :text)
    end
```